### PR TITLE
[GLUTEN-7604][CORE] Code refactors against ColumnarRuleApplier.Executor

### PR DESCRIPTION
**Pull request by beliefer**
## What changes were proposed in this pull request?

This PR proposes to give a suitable name for columnar rule executor.
This PR also contains some minor improvement.

## How was this patch tested?

integration tests

### CODE REVIEW

File: gluten-core/src/main/scala/org/apache/gluten/extension/columnar/enumerated/EnumeratedApplier.scala [hunk 1]
```diff
@@ -56,10 +56,8 @@ class EnumeratedApplier(
     finalPlan
   }
 
-  private def apply0(rules: Seq[Rule[SparkPlan]], plan: SparkPlan): SparkPlan = {
-    val executor = new ColumnarRuleApplier.Executor("ras", rules)
-    executor.execute(plan)
-  }
+  private def apply0(rules: Seq[Rule[SparkPlan]], plan: SparkPlan): SparkPlan =
+    new ColumnarRuleApplier.ColumnarRuleExecutor("ras", rules).execute(plan)
```
Comment (zhztheplayer) [inline gluten-core/src/main/scala/org/apache/gluten/extension/columnar/enumerated/EnumeratedApplier.scala:60]: `ColumnarRuleApplier.ColumnarRuleExecutor` sounds a little bit too long. Let's move `ColumnarRuleExecutor` to an individual file, i.e., `ColumnarRuleExecutor.scala`? Thanks.

File: gluten-core/src/main/scala/org/apache/gluten/extension/columnar/ColumnarRuleExecutor.scala [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    } else {
+      s"Rule $ruleName has no effect, took $millisTime ms."
+    }
+
+  override def apply(plan: SparkPlan): SparkPlan = {
+    val (out, millisTime) = GlutenTimeMetric.recordMillisTime(delegate.apply(plan))
+    logOnLevel(transformPlanLogLevel, message(plan, out, millisTime))
+    out
+  }
+}
```
Comment (zhztheplayer) [inline gluten-core/src/main/scala/org/apache/gluten/extension/columnar/ColumnarRuleExecutor.scala:62]: Thank you!

Another nit: Can we remove `LoggedRule` to `ColumnarRuleExecutor.LoggedRule`? Then it's no longer accessible from other classes.

**Review by zhztheplayer — APPROVED** [review]
Thanks!

**Review by zhztheplayer — APPROVED** [review]
Thanks!

**Comment by beliefer** [comment]
@zhztheplayer Thank you!

**Pull request merged**